### ISSUE TITLE
No apiConfig defined on module 'app'. Are you missing a module.exports?

### ISSUE THREAD

**Issue by tanepiper** [claudiajs/claudia#29]
With the following, which sits inside a folder `api/app.js`, in the package.json file for the project I have `"files": "api/*.js`

```
/*global require, module*/

var APIBuilder = require('claudia-api-builder');
var api = new APIBuilder();
var Promise = require('bluebird');
var pg = require('pg');

module.exports = api;

api.get('/app', function() {

  return new Promise(function(resolve, reject) {

    var client = new pg.Client('<connection string>');

    client.connect(function(error) {
      if (error) {
        return reject(error);
      }

      client.query('SELECT * FROM apps', function(error, results) {

        if (error) {
          return reject(error);
        }

        var output = results.map(function(result) {
          return {
              // Map return in here
          }
        });

        return resolve(output);
      });
    })
  });
});
```

I get the error in the title after running:

`claudia create --name appname --region eu-west-1 --api-module app`

The full output is:

```
server listening on port: 4001
No apiConfig defined on module 'app'. Are you missing a module.exports?
```

No `claudia.json` file is created either

I've tried to follow the example as close as possible, although this code lives in an existing project that is a static app, and rather than have a separate repo I am keeping the lamda functions within the api folder to deploy.

Any ideas?

**Comment by tanepiper** [comment]
I should also note is DOES create the Lamda, but shows no API Endpoints. It also creates the executor role, but nothing shows in the API Gateway.

**Comment by gojko** [comment]
I'm working on improving error messages now, they are not the most intuitive. in effect, this says that it can't locate the api module or cannot load it. your api module is `--api-module app`, so it's trying to require `app` and then find the apiConfig method on it. The module seems to be in a folder according to your example, so you may need to specify `--api-module api/app`

**Comment by tanepiper** [comment]
@gojko Ok cool that seemed to fix the deploy and it creates the API, but now when I go to the URL I get a stack trace in lambda:

```
{
errorMessage: "Cannot find module 'api/app'",
errorType: "Error",
stackTrace: [
"Function.Module._resolveFilename (module.js:338:15)",
"Function.Module._load (module.js:280:25)",
"Module.require (module.js:364:17)",
"require (module.js:380:17)"
]
}
```

So it doesn't seem to respecting the resolve paths for this as far as I can tell?

**Comment by tanepiper** [comment]
The claudia.json file generated is:

```
{
  "lambda": {
    "role": "myapp-executor",
    "name": "myapp",
    "region": "eu-west-1"
  },
  "api": {
    "id": "app-random-id",
    "module": "api/app"
  }
}
```

**Comment by gojko** [comment]
I've never tried to create a lambda with a function in a subfolder; this might be a limitation of the lambda nodejs engine, not sure. I'll investigate that.

**Comment by tanepiper** [comment]
@gojko OK, thanks for that.  In the meantime I'll try create it in the root folder and see how that works (as far as I can tell claudia should ignore most of the other project anyway?)

**Comment by gojko** [comment]
the rest should work ok. a trivial thing to do would be to create index.js in your root folder that just does module.exports = require('./app/api'), and you can use that as the claudia entry point, until we see how to use subfolders for api execution

**Comment by gojko** [comment]
I've added several checks to create and update that will report better on these issues and prevent common errors (https://github.com/claudiajs/claudia/commit/9d2f8b82c60977d4105411dfd7a588ea9a8907c9). From what I can see, there is no way to tell Lambda to execute a module handler in a subfolder, so the main api module has to be in root. 

this is not on NPM yet, but it should be there as 2.0 soon

**Comment by gojko** [comment]
this is now available on NPM as 1.1.0. I've made the new version fully backwards compatible, so there was no need to bump major version.
